sdk/storage/azblob/zc_append_blob_client.go

```diff
@@ -5,17 +5,40 @@ package azblob
 
 import (
 	"context"
-	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
 	"io"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
 )
 
 type AppendBlobClient struct {
 	BlobClient
 	client *appendBlobClient
 }
 
-func NewAppendBlobClient(blobURL string, cred azcore.Credential, options *ClientOptions) (AppendBlobClient, error) {
-	con := newConnection(blobURL, cred, options.getConnectionOptions())
+// NewAppendBlobClient creates an AppendBlobClient with the specified URL, Azure AD credential, and options.
+func NewAppendBlobClient(blobURL string, cred azcore.TokenCredential, options *ClientOptions) (AppendBlobClient, error) {
```

Comment:
Can we have just one `NewAppendBlobClient` interface?
```golang
func NewAppendBlobClient(blobURL string, options *CreateAppendBlobClientOptions) {
   credType, creds, clientOptions, err := options.format()
  // based on cred type, we can create clients
}
```

Inside zm_append_blob_request_options.go,
```golang
type CreateAppendBlobClientOptions struct {
     credential
    clientOptions
}

@zezha-msft , what do you think?

Reply:
Same for all the clients. We should have a uniform interface method instead of multiple methods. That was the whole reason to introduce option bags in parameters

Reply:
I don't love multiple constructors either but credential is a mandatory field, so I don't think it should fit inside the option bag.